social_core/backends/open_id_connect.py

```diff
@@ -41,6 +51,25 @@ def __init__(self, handle, secret="", issued=0, lifetime=0, assoc_type="") -> No
         self.assoc_type = assoc_type  # as state
 
 
+def generate_hash_algorithm_mapping():
+    ret = {
+        "sha256": hashlib.sha256,
+        "sha384": hashlib.sha384,
+        "sha512": hashlib.sha512,
+    }
+
+    if has_crypto:
+        ret = {
+            "sha256": hashes.SHA256,
+            "sha384": hashes.SHA384,
+            "sha512": hashes.SHA512,
+        }
+    return ret
```

Comment:
I think I've seen this in PyJWT as well, couldn't it be reused?

Reply:
There is no a plain `sha256 384 512` hash name in pyjwt but only a `es256 384 512`. I will try to find it in cryptography

Reply:
I use this solution.
```
from cryptography.hazmat.primitives import hashes

algo_class = getattr(hashes, algo_class_name, None)
```


Is it a good idea?